### PR TITLE
Implement Anisotropic Crystal example

### LINKED ISSUE #306: Implement anisotropic crystal

### PR DESCRIPTION
**Pull request by lsawade**
# Description

Implement the anisotropic crystal example from fortran specfem2d.

# Issue Number

closes #306 

# Checklist

Please make sure to check developer documentation on specfem docs. 

- [x] I ran the code through pre-commit to check style
- [x] My code passes all the integration tests
- [x] I have added sufficient unittests to test my changes
- [x] I have added/updated documentation for the changes I am proposing
- [x] I have updated CMakeLists to ensure my code builds
- [ ] My code builds across all platforms

### CODE REVIEW

File: examples/anisotropic-crystal/CMakeFiles/Snakefile.in [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+            {SPECFEM_BIN} -p {input.config} >> output.log
+        """
+
+
+rule plot_seismogram:
+    input:
+        trace_files=expand(
+            "OUTPUT_FILES/results/{station_name}{network_name}{component}.semd",
+            station_name=[
+                "S0001",
```
Comment (Rohit-Kakodkar) [inline examples/anisotropic-crystal/CMakeFiles/Snakefile.in:109]: Selectively plot only 6 seismograms

**Review by Rohit-Kakodkar — APPROVED** [review]
Minor changes. Otherwise LGTM

**Review by icui — APPROVED** [review]

File: examples/anisotropic-crystal/CMakeFiles/Snakefile.in (hunk 1 above)
Comment (lsawade) [reply examples/anisotropic-crystal/CMakeFiles/Snakefile.in]: Done!

Comment (lsawade) [reply examples/anisotropic-crystal/CMakeFiles/Snakefile.in]: Updated the plot too.

**Pull request merged**